bareos/generate_password.sls

```diff
@@ -0,0 +1,10 @@
+{% from "bareos/map.jinja" import bareos with context %}
+
+{% if salt['pillar.get']('bareos:generate_unique_password', False) %}
+bareos_password_file:
+  file.managed:
+    - name: {{ bareos.config_dir }}/bareos-dir.d/password.conf
```

Comment:
As this might be added in different hosts, not just the director, how about moving the file to, ie, `/etc/bareos/password.conf`?

Also, I'd add these parameters, to ensure permissions:

```yaml
    - mode: 640
    - user: {{ bareos.system_user }}
    - group: {{ bareos.system_group }}
```